src/blockdata/script.rs

```diff
@@ -370,7 +400,21 @@ impl Script {
     /// Compute the P2WSH output corresponding to this witnessScript (aka the "witness redeem
     /// script")
     pub fn to_v0_p2wsh(&self) -> Script {
-        Script::new_v0_wsh(&self.wscript_hash())
+        Script::new_v0_p2wsh(&self.wscript_hash())
+    }
+
+    /// Compute P2TR output with a given internal key and a single script spending path equal to the
+    /// current script, assuming that the script is a Tapscript
+    #[inline]
+    pub fn to_v1_p2tr<C: Verification>(&self, secp: &Secp256k1<C>, internal_key: UntweakedPublicKey) -> Script {
+        let leaf_hash = TapLeafHash::from_script(&self, LeafVersion::TapScript);
+        let merkle_root = TapBranchHash::from_inner(leaf_hash.into_inner());
+        Script::new_v1_p2tr(&secp, internal_key, Some(merkle_root))
+    }
+
+    #[inline]
+    fn witness_version(&self) -> Option<WitnessVersion> {
+        WitnessVersion::from_opcode(self.0[0].into()).ok()
```

Comment:
I think `self.0.get(0)?` would be better to pass the responsibility for checking to the caller.